target/snitch_cluster/sw/apps/snax-mac/src/snax-mac.c

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "snrt.h"
+
+#include "data.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    uint32_t final_output;
+
+    uint32_t *local_a, *local_b, *local_c, *local_o;
+
+    // Allocate space in TCDM
+    local_a = (uint32_t *)snrt_l1_next();
+    local_b = local_a + VEC_LEN;
+    local_c = local_b + VEC_LEN;
+    local_o = local_c + 1;
+
+    uint32_t dma_pre_load = snrt_mcycle();
+
+    if (snrt_is_dm_core()) {
+        size_t vector_size = VEC_LEN * sizeof(uint32_t);
+        size_t scale_size = 1 * sizeof(uint32_t);
+        snrt_dma_start_1d(local_a, A, vector_size);
+        snrt_dma_start_1d(local_b, B, vector_size);
+        snrt_dma_start_1d(local_c, &C, scale_size);
+    }
+
+    snrt_cluster_hw_barrier();
+
+    // Read the mcycle CSR (this is our way to mark/delimit a specific
+    // code region for benchmarking)
+    uint32_t pre_is_compute_core = snrt_mcycle();
+
+    if (snrt_is_compute_core()) {
+        // This marks the start of the accelerator style of MAC operation
+        uint32_t csr_set = snrt_mcycle();
+
+        // Set addresses
+        write_csr(0x3d0, (uint32_t)local_a);
+        write_csr(0x3d1, (uint32_t)local_b);
+        write_csr(0x3d2, (uint32_t)local_c);
+        write_csr(0x3d3, (uint32_t)local_o);
```

Comment:
Is this documented anywhere? Would be nice if we can reference to some documentation here

Reply:
It's in the notion link but will make a separate docu within the repo. I think those indeed are the next steps.

Reply:
Here is the notion link: https://danknight.notion.site/SNAX-ver-prototype-1-c0d63a16712b4f59ad7edc9dd162666c?pvs=4

Placing it here for PR documentation purposes but will move it into the git soon.